app.py

```diff
@@ -296,7 +477,123 @@ def logout():
 
+    if user == None:
+        return {
+            "error": "Invalid email: Email not found"
+        }, 404
+
+    # create a reset token and set it to be this user's reset token
+    resetToken = uuid.uuid1()
+    user.reset_token = resetToken
+    db.session.commit()
+
+    # send forgot password email to the user
+    err = forgotPasswordEmail(mail, [body['email']], user.first_name, user.last_name, str(resetToken))
+    if err != None:
+        print(err)
+        db.session.rollback()
+        return {
+             "error": "Internal Server Error"
+        }, 500
+    db.session.close()
     return {
         "success": True
     }
+
+@app.route("/resetPassword", methods=["POST"])
+def resetPassword():
+    body = request.get_json(force=True)
+    # Grab the verification token from the query parameter
+    resetToken = body['reset_token']
+    email = body['email']
+    password = body['password']
+
+    if resetToken == None:
+        return {
+            "error": "No reset_token present in query parameter"
+        }, 400
+    if email == None:
+        return {
+            "error": "No email present in query parameter"
+        }, 400
+
+    # Check that the verification_token belongs to the email
+    user = User.query.filter_by(email=email, reset_token=resetToken).first()
+    if user == None:
+        return {
+            "error": "Invalid reset_token or email"
+        }, 404
+
+    # If it does, then we set the verified field to True and remove the verification_token
+    
+    user.reset_token = ''
+    # Encrypt the password
+    encodedPassword = bcrypt.generate_password_hash(password).decode(encoding="utf-8")
+    user.password = encodedPassword
+    db.session.commit()
+    db.session.close()
+
+    # Redirect to the frontend homepage
+    login = os.getenv("FRONTEND_URL") + 'auth/login'
+    return redirect(login, code=302)
+
+# terminate client endpoint
+@app.route("/terminateClient", methods=["PUT"])
+@http_guard(renew=True, nullable=False)
+def terminateClient(token_claims):
+    body = request.get_json(force=True)
+    # Check that the role of the requestee is COACH
+    if token_claims['role'] != Role.COACH.name:
+        return {
+            "error": "Expected role of COACH"
+    }, 400
+
+    # retrieve user with id passed in
+    user = User()
+    user = User.query.get(body['id'])
+
+    try:
+        # update the approved field for this user to null
+        user.approved = None
+        db.session.commit()
+    except Exception as e:
+        return {
+            "error": body['id']
+        }, 500
+        raise
+
+    # Grab the user from the database and dump the result into a user schema
+    user = User.query.get(user.id)
+    result = user_schema.dump(user)
+    # remove the sensitive data fields
+    del result['password']
+    del result['access_token']
+    del result['verification_token']
+
+    db.session.close()
+    # Return the user
+    return {
+        "Approved": result
```

Comment:
```suggestion
        "user": result
```
Probably better to change this to user, since the user isn't being approved